crawler/clearnet_crawler.py

```
@@ -229,3 +237,47 @@ async def _rate_limit(self, domain: str) -> None:
+            return False
+
+        base = f"{parsed.scheme}://{parsed.netloc}"
+        now = time.time()
+
+        async with self._lock:
+            cached = self._robots_cache.get(base)
+
+        if cached and (now - cached[0]) < ROBOTS_CACHE_TTL:
+            try:
+                return cached[1].can_fetch(HEADERS["User-Agent"], url)
+            except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)